src/rust/engine/process_execution/remote/src/remote_cache.rs

```diff
@@ -65,6 +74,44 @@ pub struct RemoteCacheProviderOptions {
   pub rpc_timeout: Duration,
 }
 
+async fn choose_provider(
+  options: RemoteCacheProviderOptions,
+) -> Result<Arc<dyn ActionCacheProvider>, String> {
+  let address = options.action_cache_address.clone();
+
+  // TODO: we shouldn't need to gin up a whole copy of this struct; it'd be better to have the two
+  // set of remoting options managed together.
+  let remote_options = RemoteOptions {
+    cas_address: address.clone(),
+    instance_name: options.instance_name.clone(),
+    headers: options.headers.clone(),
+    tls_config: tls::Config::new_without_mtls(options.root_ca_certs.clone()),
+    rpc_timeout: options.rpc_timeout,
+    rpc_concurrency_limit: options.concurrency_limit,
+    // TODO: these should either be passed through or not synthesized here
+    chunk_size_bytes: 0,
+    rpc_retries: 0,
+    capabilities_cell_opt: None,
+    batch_api_size_limit: 0,
+  };
+
+  if REAPI_ADDRESS_SCHEMAS.iter().any(|s| address.starts_with(s)) {
+    Ok(Arc::new(reapi::Provider::new(options).await?))
+  } else if let Some(path) = address.strip_prefix("file://") {
+    // supporting local "file://" for a 'remote' store is a bit weird... but this is handy for
+    // testing
```

Comment:
Nit: please use complete sentences for comments: https://www.pantsbuild.org/docs/style-guide#style